client/src/panel-toc-editor.ts

```diff
@@ -99,129 +204,136 @@ export const showTocEditor = (panelType: PanelType, resourceRootDir: string, act
   panel.reveal(vscode.ViewColumn.One)
   activePanelsByType[panelType] = panel
 
-  const messageQueued: {uneditable: TocTreeCollection[], editable: TocTreeCollection[]} = {
-    uneditable: [],
-    editable: []
-  }
-  const uri = getRootPathUri()
-  if (uri != null) {
-    const collectionFiles = fs.readdirSync(path.join(uri.fsPath, 'collections'))
-    const collectionTrees = []
-    for (const collectionFile of collectionFiles) {
-      const collectionData = fs.readFileSync(path.join(uri.fsPath, 'collections', collectionFile), { encoding: 'utf-8' })
-      collectionTrees.push(parseCollection(collectionData))
-    }
-    // Some special non-editable collections
-    const allModules = fs.readdirSync(path.join(uri.fsPath, 'modules'))
-    const usedModules: string[] = []
-    for (const collectionTree of collectionTrees) {
-      insertUsedModules(usedModules, collectionTree)
-    }
-    const usedModulesSet = new Set(usedModules)
-    const orphanModules = allModules.filter(x => !usedModulesSet.has(x))
-    const collectionAllModules: TocTreeCollection = {
-      type: 'collection',
-      title: 'All Modules',
-      slug: 'mock-slug__source-only',
-      children: allModules.map(moduleObjectFromModuleId).sort((m, n) => m.moduleid.localeCompare(n.moduleid))
-    }
-    const collectionOrphanModules: TocTreeCollection = {
-      type: 'collection',
-      title: 'Orphan Modules',
-      slug: 'mock-slug__source-only',
-      children: orphanModules.map(moduleObjectFromModuleId).sort((m, n) => m.moduleid.localeCompare(n.moduleid))
-    }
+  panel.webview.onDidReceiveMessage(ensureCatch(handleMessage(panel)))
 
-    // messageQueued reference passed to handleMessage so we must mutate
-    // the original object instead of setting messageQueued to a new value
-    messageQueued.uneditable = [collectionAllModules, collectionOrphanModules]
-    messageQueued.editable = collectionTrees
-  }
-
-  panel.webview.onDidReceiveMessage(ensureCatch(handleMessage(panel, messageQueued)))
+  // Setup an occasional refresh to remain reactive to unhandled events
+  const autoRefresh = setInterval(() => {
+    handleMessage(panel)({ type: 'refresh' })
+      // eslint-disable-next-line node/handle-callback-err
+      .catch(err => {
+        // Panel was probably disposed
+        clearInterval(autoRefresh)
+      })
+  }, 1000)
```

Comment:
This may just be when I run the extension locally on my system, but I noticed when I click on a dropdown menu in the ToC editor, it will flicker in and out at this interval 🤷‍♂️

Reply:
I've attempted to fix some of the flickering issues. I couldn't reproduce the dropdown flickering (but I found one with typing for renaming as well). Mind checking again?

Reply:
Yep, the change seems to have fixed what I was seeing earlier 🎉